src/shared/install.c

```diff
@@ -1449,14 +1462,68 @@ static int unit_file_search(
                         if (r >= 0) {
                                 info->path = path;
                                 path = NULL;
-                                return r;
+                                result = r;
+                                found_unit = true;
+                                break;
                         } else if (!IN_SET(r, -ENOENT, -ENOTDIR, -EACCES))
                                 return r;
                 }
         }
 
-        log_debug("Cannot find unit %s%s%s.", info->name, template ? " or " : "", strempty(template));
-        return -ENOENT;
+        if (!found_unit) {
+                log_debug("Cannot find unit %s%s%s.", info->name, template ? " or " : "", strempty(template));
+                return -ENOENT;
+        }
+
+        /* Search for drop-in directories */
+
+        dropin_dir_name = strjoina(info->name, ".d");
+        STRV_FOREACH(p, paths->search_path) {
+                char *path = NULL;
+
+                path = path_join(paths->root_dir, *p, dropin_dir_name);
+                if (!path)
+                        return -ENOMEM;
+
+                r = strv_consume(&dirs, path);
+                if (r < 0) {
+                        return r;
+                }
+        }
+
+        if (template) {
+                dropin_template_dir_name = strjoina(template, ".d");
+                STRV_FOREACH(p, paths->search_path) {
+                        char *path = NULL;
+
+                        path = path_join(paths->root_dir, *p, dropin_template_dir_name);
+                        if (!path)
+                                return -ENOMEM;
+
+                        r = strv_consume(&dirs, path);
+                        if (r < 0) {
+                                return r;
+                        }
+                }
+        }
+
+        /* Load drop-in conf files */
+
+        r = conf_files_list_strv(&files, ".conf", NULL, 0, (const char**) dirs);
+        if (r < 0) {
+                log_error_errno(r, "Failed to get list of conf files: %m");
+                return r;
```

Comment:
`log_error_errno returns` `r`, so `return log_errno_errno(r, ...);`.
Also, this is a "library" function. Above, all log statements were at debug level. You need to the same in the part you add: either use log_debug or nothing at all.

Reply:
Excuse me, I don't understand. I see `unit_file_load()` contains `log_error_errno` too. Does that mean `unit_file_load()` is not a library function?

Reply:
The general rule is that any specific function should either always log or never log. This way we can arrange things that we always get an error message, but not more than one. Generally it's nice if the caller outputs the logs, but sometimes it's not possible because the caller does not have enough information for a truly useful error message. We are not very consistent in this unfortunately.

Reply:
I see. Thanks for your helpful explanation.